utils/firebase/auth.ts

```diff
@@ -0,0 +1,32 @@
+import { signInWithPopup } from "firebase/auth";
+import { setDoc, doc, getDoc } from "firebase/firestore";
+import { UserDoc } from "models/users";
+import { auth, authProvider, userRef } from "./init";
+
+export const signin = async () => {
+    try {
+        const { user } = await signInWithPopup(auth, authProvider);
```

Comment:
Does this just trigger like a Google sign-in window? What happens if the user doesn't have a Google account?

Reply:
Per our discussion with Gracie today, we should add an alternative form of auth besides Gmail but we don't have to do it in this PR.

Reply:
Working on that currently!